tests/net/mld/src/main.c

```diff
@@ -547,21 +552,19 @@ static void test_solicit_node(void)
 			"solicit node multicast address");
 }
 
-void test_main(void)
+ZTEST(net_mld_test_suite, test_net_mld)
 {
-	ztest_test_suite(net_mld_test,
-			 ztest_unit_test(test_mld_setup),
-			 ztest_unit_test(test_join_group),
-			 ztest_unit_test(test_leave_group),
-			 ztest_unit_test(test_catch_join_group),
-			 ztest_unit_test(test_catch_leave_group),
-			 ztest_unit_test(test_verify_join_group),
-			 ztest_unit_test(test_verify_leave_group),
-			 ztest_unit_test(test_catch_query),
-			 ztest_unit_test(test_verify_send_report),
-			 ztest_unit_test(test_allnodes),
-			 ztest_unit_test(test_solicit_node)
-			 );
-
-	ztest_run_test_suite(net_mld_test);
+	test_mld_setup();
```

Comment:
I think that putting all tests under a single test case might be confusing if something fails - you don't get now a clear information on which tests case failed in the report.

Reply:
As some of the test cases are linked together, for example join group and leave group, I have combined them into groups and make them a test case.